flake.nix

```diff
@@ -58,18 +63,35 @@
         };
       };
 
-      devShells.x86_64-linux.default = (project pkgsLocal).shellFor {
-        inherit (checks.x86_64-linux.pre-commit-check) shellHook;
-        withHoogle = false;
-        buildInputs =
-          with pkgsLocal;
-          [
-            haskellPackages.cabal-fmt
-            nixpkgs-fmt
-            nodejs
-          ]
-          ++ checks.x86_64-linux.pre-commit-check.enabledPackages;
-      };
+      devShells = forAllSystems (system:
+        let
+          pkgs = pkgsFor system;
+          proj = project pkgs;
+
+          # Only x86_64-linux has pre-commit checks defined
+          shellHook = if system == "x86_64-linux"
+                      then checks.x86_64-linux.pre-commit-check.shellHook
+                      else "";
+
+          # Only x86_64-linux has pre-commit packages
+          preCommitPackages = if system == "x86_64-linux"
+                             then checks.x86_64-linux.pre-commit-check.enabledPackages
+                             else [];
```

Comment:
I don't think we need this, because the pre-commit script encodes absolute `/nix/store/...` paths to the binaries it wants.

And for future reference, this sort of condition exists in nixpkgs as [`lib.optional`](https://noogle.dev/f/lib/optional).